astropy/io/fits/connect.py

```diff
@@ -302,3 +306,202 @@ def write_table_fits(input, output, overwrite=False):
 io_registry.register_reader('fits', Table, read_table_fits)
 io_registry.register_writer('fits', Table, write_table_fits)
 io_registry.register_identifier('fits', Table, is_fits)
+
+
+def read_data_fits(filename, ext_data=0, ext_meta=0, ext_mask='mask',
+                   ext_uncert='uncert', kw_unit='bunit', copy=False,
+                   dtype=None, **kwargs_for_open):
+    """
+    Read data from a FITS file and wrap the contents in a \
+    `~astropy.nddata.NDData`.
+
+    Parameters
+    ----------
+    filename : str and other types
+        see :func:`~astropy.io.fits.open` what possible types are allowed.
+
+    ext_data, ext_meta, ext_mask, ext_uncert : str or int, optional
+        Extensions from which to read ``data``, ``meta``, ``mask`` and
+        ``uncertainty``.
+        Default is ``0`` (data), ``0`` (meta), ``'mask'`` (mask) and
+        ``'uncert'`` (uncertainty).
+
+    kw_unit : str or None, optional
+        The header keyword which translates to the unit for the data. Set it
+        to ``None`` if parsing the unit results in a ValueError during reading.
+        Default is ``'bunit'``.
+
+    copy : bool, optional
```

Comment:
I'm not sure anymore. It was either because I thought using a `copy` would solve the open-files issue or because I hadn't hardcoded the `open-mode`.

But the `copy` can go, I'll remove it.